apps/studio/pages/project/[ref]/logs/explorer/index.tsx

```diff
@@ -208,20 +306,44 @@ export const LogsExplorerPage: NextPageWithLayout = () => {
             onClear={handleClear}
             hasEditorValue={Boolean(editorValue)}
             templates={TEMPLATES.filter((template) => template.mode === 'custom')}
+            warehouseCollections={warehouseCollections || []}
             onSelectTemplate={onSelectTemplate}
+            warehouseTemplates={createWarehouseQueryTemplates(warehouseCollections || [])}
+            onSelectWarehouseTemplate={(template) => {
+              setWarehouseEditorValue(template.query)
+              setWarehouseEditorId(uuidv4())
+            }}
             onSave={handleOnSave}
             isLoading={isLoading}
             warnings={warnings}
+            dataSource={sourceType}
+            onDataSourceChange={(srcType) => {
+              setSourceType(srcType)
+              router.push({
+                pathname: router.pathname,
+                query: { ...router.query, source: srcType },
+              })
+            }}
           />
 
           <ShimmerLine active={isLoading} />
-          <CodeEditor
-            id={editorId}
-            language="pgsql"
-            defaultValue={editorValue}
-            onInputChange={(v) => setEditorValue(v || '')}
-            onInputRun={handleRun}
-          />
+          {sourceType === 'warehouse' ? (
+            <CodeEditor
```

Comment:
Command Enter doesn't run the code for warehouse code editor

Reply:
Not sure if BE or FE issue but when I do try to run the query by clicking the Run button, it's saying that i don't have a collection of that name, although I do
![image](https://github.com/supabase/supabase/assets/19742402/9b04e75c-d768-42f3-859f-853ac940aa77)